wallet/src/repository/wallet/mod.rs

```diff
@@ -915,13 +915,28 @@ where
             value,
             fee,
             time_lock,
+            fee_type,
         }: types::VttParams,
     ) -> Result<types::VTTransaction> {
         let mut state = self.state.write()?;
-        let components =
-            self.create_vt_transaction_components(&mut state, value, fee, Some((pkh, time_lock)))?;
-
-        let body = types::VTTransactionBody::new(components.inputs, components.outputs);
+        let (body, components) = match fee_type {
+            types::FeeType::Weighted => {
+                let components = self.create_vt_transaction_components(
+                    &mut state,
+                    value,
+                    fee,
+                    Some((pkh, time_lock)),
+                )?;
+                let body = types::VTTransactionBody::new(
+                    components.inputs.clone(),
+                    components.outputs.clone(),
+                );
+                (body, components)
+            }
+            types::FeeType::Absolute => {
+                self.create_vtt_body(&mut state, value, fee, Some((pkh, time_lock)))?
```

Comment:
Isn't this backwards? Maybe we can rename the function to make it clear that it's the same as `create_vt_transaction_components` but with a weighted fee.
```suggestion
            types::FeeType::Weighted => {
                self.create_vt_components_weighted_fee(&mut state, value, fee, Some((pkh, time_lock)))?
```

Reply:
GOod

Reply:
https://github.com/witnet/witnet-rust/issues/1697